### PR TITLE
feat: add camera as entity

### LINKED ISSUE #36: Camera as entity

### PR DESCRIPTION
**Pull request by Mathieu-Lala**

### CODE REVIEW

File: src/KawaiiEngine/include/Engine.hpp [hunk 1]
```diff
@@ -273,36 +282,21 @@ class Engine {
...
             });
         }
 
+        // note : is it the best way of doing that ?
+        for (const auto &entity : world.view<Velocity3f, CameraData>()) {
+            const auto &vel = world.get<Velocity3f>(entity);
+            world.patch<CameraData>(entity, [&vel, &dt_secs](auto &cam) {
+                cam.target_center += vel.component * static_cast<double>(dt_secs);
+            });
+        }
```
Comment (Mathieu-Lala) [inline src/KawaiiEngine/include/Engine.hpp:298]: Target_center should be a child entity

Comment (ltabis) [reply src/KawaiiEngine/include/Engine.hpp:298]: why ?

**Review by ltabis — APPROVED** [review]
All good.

File: src/KawaiiEngine/include/Engine.hpp (hunk 1 above)
Comment (Mathieu-Lala) [reply src/KawaiiEngine/include/Engine.hpp:298]: If it is a separated entity (or a child even better), and have his own position, then the system velocity-position will be trigger automatically and we don t need this very specific system

**Pull request merged**